src/Passage.sol

```diff
@@ -104,83 +69,19 @@ contract Passage is BasePassage {
-    /// @param tokenOut - The address of the token the user expects to receive on the target chain.
-    /// @param recipient - The address of the recipient of tokenOut on the target chain.
-    /// @param deadline - The deadline by which the swap order must be fulfilled.
-    /// @param amountIn - The amount of tokenIn the user supplies as the input on the rollup for the trade.
-    /// @param amountOut - The minimum amount of tokenOut the user expects to receive on the target chain.
-    /// @custom:reverts Expired if the deadline has passed.
-    /// @custom:emits Swap if the swap transaction succeeds.
-    function swap(
-        uint256 targetChainId,
-        address tokenIn,
-        address tokenOut,
-        address recipient,
-        uint256 deadline,
-        uint256 amountIn,
-        uint256 amountOut
-    ) external payable {
-        // check that the deadline hasn't passed
-        if (block.timestamp >= deadline) revert OrderExpired();
-
-        if (tokenIn == address(0)) {
-            require(amountIn == msg.value);
-        } else {
-            IERC20(tokenIn).transferFrom(msg.sender, address(this), amountIn);
-        }
-
-        // emit the swap event
-        emit Swap(targetChainId, tokenIn, tokenOut, recipient, deadline, amountIn, amountOut);
-    }
-
-    /// @notice Transfer the entire balance of ERC20 tokens to the recipient.
-    /// @dev Called by the Builder within the same block as users' `swap` transactions
-    ///      to claim the amounts of `tokenIn`.
-    /// @dev Builder MUST ensure that no other account calls `sweep` before them.
-    /// @param token - The token to transfer.
-    /// @param recipient - The address to receive the tokens.
-    function sweep(address token, address recipient) public {
-        uint256 balance;
-        if (token == address(0)) {
-            balance = address(this).balance;
-            payable(recipient).transfer(balance);
-        } else {
-            balance = IERC20(token).balanceOf(address(this));
-            IERC20(token).transfer(recipient, balance);
-        }
-        emit Sweep(token, recipient, balance);
+/// @notice A contract deployed to the Rollup that allows users to remove tokens from the Rollup TVL back to the Host.
+contract RollupPassage {
+    /// @notice Emitted when tokens exit the rollup.
+    /// @param token - The address of the token exiting the rollup.
+    /// @param recipient - The desired recipient of the token on the host chain.
+    /// @param amount - The amount of the token entering the rollup.
+    event Exit(address indexed token, address indexed recipient, uint256 amount);
+
+    /// @notice Allows native Ether to exit the rollup.
+    /// @dev Rollup node will burn the msg.value.
+    /// @param recipient - The desired recipient of the Ether on the host chain.
+    /// @custom:emits Exit indicating the amount of Ether to burn on the rollup & the recipient on the host chain.
+    function exit(address recipient) public payable {
```

Comment:
probably out of scope for this PR as it's unspecified new behavior in a refactor